src/analysis/convex/cone.lean

```diff
@@ -643,4 +643,23 @@ begin
       set.mem_singleton_iff, forall_eq, imp_self] }
 end
 
+lemma closed_inner_dual_cone : is_closed (s.inner_dual_cone : set H) :=
+begin
+  -- reduce the problem to showing that dual cone of a singleton `{x}` is closed
+  rw inner_dual_cone_eq_Inter_inner_dual_cone_singleton,
+  apply is_closed_Inter,
+  simp_rw [set_coe.forall, subtype.coe_mk],
+  intros x hx,
+
+  -- the dual cone of a singleton set is the preimage of `[0, ∞)` under `inner x`
+  have h : (({x} : set H).inner_dual_cone : set H) = (inner x : H → ℝ)⁻¹' (set.Ici 0),
+  { apply set.ext,
+    simp only [convex_cone.mem_coe, mem_inner_dual_cone, mem_singleton_iff, forall_eq, mem_preimage,
+    mem_Ici, iff_self, forall_const] },
+
+  -- the preimage is closed as `inner x` is continuous and `[0, ∞)` is closed
+  rw h,
+  exact is_closed.preimage (continuous.inner continuous_const continuous_id) is_closed_Ici,
```

Comment:
```suggestion
  exact is_closed_Ici.preimage (continuous.inner continuous_const continuous_id),
```

Reply:
there might be even the point of marking `continuous.inner` with the `continuity` flag and then the proof would be
`exact is_closed_Ici.preimage (by continuity)`

Reply:
> One note for reviewing: if you a addressed the comment, you should click on "resolve conversation" so that only the active discussions are open.

Thanks, will do!